sno/init.py

```diff
@@ -577,51 +575,53 @@ def import_table(
 
     use_repo_ctx = not do_list
     if use_repo_ctx:
-        repo_path = ctx.obj.repo_path
         repo = ctx.obj.repo
         check_git_user(repo)
 
-    source_loader = OgrImporter.open(source, table)
+    source_loader = OgrImporter.open(source, None)
 
     if do_list:
         source_loader.print_table_list(do_json=output_format == 'json')
         return
-
-    source_loader.check_table(prompt=True)
-
-    if directory:
-        directory = os.path.relpath(directory, os.path.abspath(repo_path))
-        if not directory:
-            raise click.BadParameter("Invalid import directory", param_hint="directory")
-        if is_windows:
-            directory = directory.replace("\\", "/")  # git paths use / as a delimiter
+    elif all_tables:
+        tables = source_loader.get_tables().keys()
     else:
-        directory = source_loader.table
+        if not tables:
+            tables = [source_loader.prompt_for_table("Select a table to import")]
+
+    loaders = {}
+    for table in tables:
+        (src_table, *rest) = table.split(':', 1)
+        dst_table = rest[0] if rest else src_table
+        if not dst_table:
+            raise click.BadParameter("Invalid table name", param_hint="tables")
+        if is_windows:
+            dst_table = dst_table.replace("\\", "/")  # git paths use / as a delimiter
 
-    importer = structure.DatasetStructure.importer(directory, version=version)
-    params = json.loads(os.environ.get("SNO_IMPORT_OPTIONS", None) or "{}")
-    if params:
-        click.echo(f"Import parameters: {params}")
-    importer.fast_import_table(repo, source_loader, message=message, **params)
+        if dst_table in loaders:
+            raise InvalidOperation(
```

Comment:
Although there's no rule for deciding this, maybe this should be a click.UsageError